React/Modules/RCTTiming.m

```diff
@@ -181,22 +192,31 @@ - (void)didUpdateFrame:(__unused RCTFrameUpdate *)update
   // Call timers that need to be called
   if (timersToCall.count > 0) {
     [_bridge enqueueJSCall:@"JSTimersExecution.callTimers" args:@[timersToCall]];
-
-    // If we call at least one timer this frame, don't switch to a paused state yet, so if
-    // in response to this timer another timer is scheduled, we don't pause and unpause
-    // the displaylink frivolously.
-    return;
   }
 
-  // No need to call the pauseCallback as RCTDisplayLink will ask us about our paused
-  // status immediately after completing this call
-  if (_timers.count == 0) {
-    _paused = YES;
+  if (_sendIdleEvents) {
+    NSTimeInterval frameElapsed = (CACurrentMediaTime() - update.timestamp) * 1000;
+    NSTimeInterval currentTimestamp = [[NSDate date] timeIntervalSince1970] * 1000;
+    NSNumber *absoluteFrameStartMillis = [NSNumber numberWithDouble:currentTimestamp - frameElapsed];
+    if (kFrameDuration - frameElapsed >= kIdleCallbackFrameDeadline) {
+      [_bridge enqueueJSCall:@"JSTimersExecution.callIdleCallbacks" args:@[absoluteFrameStartMillis]];
+    }
   }
-  // If the next timer is more than 1 second out, pause and schedule an NSTimer;
-  else if ([nextScheduledTarget timeIntervalSinceNow] > kMinimumSleepInterval) {
-    [self scheduleSleepTimer:nextScheduledTarget];
-    _paused = YES;
+
+  // If we call at least one timer this frame, don't switch to a paused state yet, so if
```

Comment:
Since the if-statement's logic was reversed, maybe rephrase this as:

> Switch to a paused state *only* if we didn't call at least one timer this frame, so if...